### ISSUE TITLE
"ReferenceError: window is not defined" when using an Space API with @gradio/client in Node.js

### ISSUE THREAD

**Issue by L3-N0X** [gradio-app/gradio#5774]
### Describe the bug

I'm using the gradio client in Node.js to make API calls to Spaces on Huggingface [Project: Whatsapp-web-js Bot which uses spaces to answer on specific messages], but an error appears when I'm trying to get predictions. The error is `ReferenceError: window is not defined` and I think the line causing this issue is [this one](https://github.com/gradio-app/gradio/blob/2626a26f187ade17a139cea7435241585ff04de3/client/js/src/client.ts#L423C28-L423C28). I tried fixing it myself (Removing everything with url_params) which worked temporarily, but now I want to deploy my project and I can't change something in the npm files.

Is there a workaround for this or something?

### Have you searched existing issues?  🔎

- [x] I have searched and found no existing issues

### Reproduction

Use Node.js and a space via API and use any endpoint

```js
import { client } from "@gradio/client";
const app = await client("illia56/llama-2-7b-chat");
const result = await app.predict("/api", [
  "Warum ist die Banane krumm?",
]);
console.log(result.data);
```

### Logs

```shell
`window.location.search
          ^

ReferenceError: window is not defined
    at submit (file:///c:/LEON/CODING/JS/wa-bot-temp/node_modules/@gradio/client/dist/index.js:412:11)
    at file:///c:/LEON/CODING/JS/wa-bot-temp/node_modules/@gradio/client/dist/index.js:368:23
    at new Promise (<anonymous>)
    at Object.predict (file:///c:/LEON/CODING/JS/wa-bot-temp/node_modules/@gradio/client/dist/index.js:367:16)
    at file:///c:/LEON/CODING/JS/wa-bot-temp/index.js:3:26
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)

Node.js v20.6.0`
```

### System Info

```shell
Node.js v20.6.0
@gradio/client 0.4.1
```

### Severity

I can work around it

**Comment by abidlabs** [comment]
Hi @L3-N0X would you like to open a PR with your change and we can take a look?

**Comment by L3-N0X** [comment]
Yes, I'll make an PR

**Comment by gibiee** [comment]
I made a PR #5787 about this problem. I hope it will be helpful.

**Comment by freddyaboulton** [comment]
Closing as fixed by #5787! Thanks @gibiee !
